data/shared/src/main/scala/sigma/serialization/MethodCallSerializer.scala

```diff
@@ -23,6 +25,9 @@ case class MethodCallSerializer(cons: (Value[SType], SMethod, IndexedSeq[Value[S
     w.putValue(mc.obj, objInfo)
     assert(mc.args.nonEmpty)
     w.putValues(mc.args, argsInfo, argsItemInfo)
+    if(mc.method.name == SGlobalMethods.deserializeRawMethod.name){
+      w.putType(mc.tpe, typeInfo)
+    }
```

Comment:
Better leave MethodCall unchanged and use PolyMethodCall as described in https://github.com/ScorexFoundation/sigmastate-interpreter/issues/980
This will also give both simplicity of code and efficiency of execution, because specialization will happen at compile time and most of the method calls are simple method calls.